File: scripts/check_album_ids.js
const fs = require('fs')
const g = JSON.parse(fs.readFileSync('data/groups.json','utf8'))
const re = /^[A-Za-z0-9]+$/
let bad = []
for (const gr of g) {
  const gid = gr.id
  const gname = gr.name
  const aid = gr.spotify_album_id
  if (aid) {
    const norm = aid.split('?')[0]
    if (!re.test(norm)) bad.push([gid,gname,'spotify_album_id',aid])
  }
  const s = gr.spotify || {}
  for (const a of (s.albums || [])) {
    const id = a.id
    if (id && !re.test(id)) bad.push([gid,gname,'spotify.albums.id',id])
  }
  for (const a of (gr.albums || [])) {
    const id = a.id
    if (id && !re.test(id)) bad.push([gid,gname,'albums.id',id])
  }
}
if (!bad.length) console.log('OK: no invalid-looking album ids found')
else bad.forEach(b => console.log(b.join('|')))

File: scripts/check_album_ids.py
import json, re, sys
p = 'data/groups.json'
with open(p, 'r', encoding='utf8') as f:
    groups = json.load(f)

bad = []
for g in groups:
    gid = g.get('id')
    gname = g.get('name')
    aid = g.get('spotify_album_id')
    if aid:
        norm = aid.split('?')[0]
        if not re.match(r'^[A-Za-z0-9]+$', norm):
            bad.append((gid, gname, 'spotify_album_id', aid))
    s = g.get('spotify') or {}
    for a in s.get('albums', []):
        aid = a.get('id')
        if aid and not re.match(r'^[A-Za-z0-9]+$', aid):
            bad.append((gid, gname, 'spotify.albums.id', aid))
    for a in g.get('albums', []):
        aid = a.get('id')
        if aid and not re.match(r'^[A-Za-z0-9]+$', aid):
            bad.append((gid, gname, 'albums.id', aid))

if not bad:
    print('OK: no invalid-looking album ids found')
else:
    for b in bad:
        print('|'.join(map(str,b)))
    sys.exit(1)
